src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenThatWeWantToGetDependenciesViaDesignTimeBuild.cs

```diff
@@ -0,0 +1,839 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using FluentAssertions;
+using Microsoft.Build.Framework;
+using Xunit;
+using static Microsoft.NET.Build.Tasks.ResolvePackageAssets;
+
+namespace Microsoft.NET.Build.Tasks.UnitTests
+{
+    public class GivenThatWeWantToGetDependenciesViaDesignTimeBuild
+    {
+        [Fact]
+        public void ItShouldIgnoreAllDependenciesWithTypeNotEqualToPackageOrUnresolved()
+        {
+            string projectAssetsJsonPath = Path.GetTempFileName();
+            string projectCacheAssetsJsonPath = Path.GetTempFileName();
```

Comment:
The test should delete both of these files when it completes, to avoid polluting the temp dir.